### PR TITLE
chore: DAH-2734 removing hardcoded fcfs checks

### PR DESCRIPTION
**Pull request by alulabeshue-sfgov**
## Description

Remove the hardcoded checks for fcfs, instead use the data coming from salesforce

## Jira ticket

[DAH-2734](https://sfgovdt.jira.com/browse/DAH-2734)

### Version Control

- [ ] branch name begins with `angular` if it contains updates to Angular code
- [x] branch name contains the Jira ticket number
- [x] PR name follows `type: TICKET-NUMBER Description` format, e.g. `feat: DAH-123 New Feature`

### Code quality

- [x] [the set of changes is small](https://google.github.io/eng-practices/review/developer/small-cls.html#what-is-small)
- [x] all automated code checks pass (linting, tests, coverage, etc.)
- [x] code irrelevant to the ticket is not modified e.g. changing indentation due to automated formatting
- [x] if the code changes the UI, it matches the UI design exactly

### Review instructions

- [x] instructions specify which environment(s) it applies to
- [x] instructions work for PA testers
- [x] instructions have already been performed at least once

### Request review

- [x] PR has `needs review` label
- [x] Use `Housing Eng` group to automatically assign reviewers, and/or assign specific engineers
- [ ] If time sensitive, notify engineers in Slack

[DAH-2734]: https://sfgovdt.jira.com/browse/DAH-2734?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/javascript/modules/listingDetailsAside/ListingDetailsApplicationDate.tsx [hunk 1]
```diff
@@ -54,9 +54,9 @@ const getStatusLottery = (listing: RailsListing) => {
 }
 
 const getStatusFcfs = (listing: RailsListing) => {
-  const isClosed = listing.Listing_State === LISTING_STATES.CLOSED
+  const isClosed = !listing.Accepting_Online_Applications
   const datetime = listing.Application_Start_Date_Time
-  const showDateTime = listing.Listing_State === LISTING_STATES.NOT_YET_OPEN
```
Comment (jimlin-sfgov) [inline app/javascript/modules/listingDetailsAside/ListingDetailsApplicationDate.tsx:59]: question: can we remove `Listing_State` from the `RailsListing` type? Does it actually exist in the data from the Salesforce API?

**Review by jimlin-sfgov — COMMENTED** [review]
From following the review instructions I see discrepancies with the Figma designs, is following the designs within the scope of this ticket?

Accepting Online Applications is checked AND Application Start Date/Time in the future:
button says 'learn more and apply',  figma button says 'get started'
Screenshot 2024-10-04 at 4 22 31 PM

Accepting Online Applications not checked:
Lottery and Lottery info is present, figma doesn't have these
Screenshot 2024-10-04 at 4 34 34 PM

**Comment by alulabeshue-sfgov** [comment]
> From following the review instructions I see discrepancies with the Figma designs, is following the designs within the scope of this ticket? Are the designs outdated?

@jimlin-sfgov thanks for testing! 

For the lottery info and lottery results section this is based on how the listing is setup in salesforce. Im not sure which listing you are using but it looks like it is a fcfs listing that has lottery information in salesforce. I've brought this up to Bridget and for v1 of this feature it will be handled by training, SMEs will be trained to avoid running a lottery or adding lottery information to a fcfs listing in salesforce. The scope of this ticket is to focus on the how to apply section and the status sections on the details and directory pages.

For the how to apply button text the design changed after we played [DAH-2493](https://sfgovdt.jira.com/browse/DAH-2493), there is a follow up [ticket](https://sfgovdt.jira.com/browse/DAH-2740) to update this text. I've updated the review instructions.

**Comment by jimlin-sfgov** [comment]
> Im not sure which listing you are using but it looks like it is a fcfs listing that has lottery information in salesforce.

I'm using the listing in the review instructions (`/listings/a0W4U00000Nlb5lUAB`). Perhaps you should also update the instructions about the lottery information, for the PA tester.

**Review by jimlin-sfgov — APPROVED** [review]
LGTM

The UI discrepancies are expected.

**Pull request merged**